pkg/cmd/search/repos/repos.go

```diff
@@ -0,0 +1,198 @@
+			opts.Query.Keywords = args
+			if opts.Query.Limit < 1 || opts.Query.Limit > 1000 {
+				return cmdutil.FlagErrorf("`--limit` must be between 1 and 1000")
+			}
+			if runF != nil {
+				return runF(opts)
+			}
+			return reposRun(opts)
+		},
+	}
+
+	// Output flags
+	cmdutil.AddJSONFlags(cmd, &opts.Exporter, search.RepositoryFields)
+	cmd.Flags().BoolVarP(&opts.WebMode, "web", "w", false, "Open the search query in the web browser")
+
+	// Query parameter flags
+	cmd.Flags().IntVarP(&opts.Query.Limit, "limit", "L", 30, "Maximum number of repositories to fetch")
+	cmdutil.StringEnumFlag(cmd, &opts.Query.Order, "order", "", "", []string{"asc", "desc"}, "Order of repositories returned, ignored unless '--sort' flag is specified")
+	cmdutil.StringEnumFlag(cmd, &opts.Query.Sort, "sort", "", "", []string{"forks", "help-wanted-issues", "stars", "updated"}, "Sort fetched repositories")
+
+	// Query qualifier flags
+	cmdutil.NilBoolFlag(cmd, &opts.Query.Qualifiers.Archived, "archived", "", "Filter based on archive state")
+	cmd.Flags().StringVar(&opts.Query.Qualifiers.Created, "created", "", "Filter based on created at date")
+	cmd.Flags().StringVar(&opts.Query.Qualifiers.Followers, "followers", "", "Filter based on number of followers")
+	cmdutil.StringEnumFlag(cmd, &opts.Query.Qualifiers.Fork, "include-forks", "", "", []string{"false", "true", "only"}, "Include forks in fetched repositories")
+	cmd.Flags().StringVar(&opts.Query.Qualifiers.Forks, "forks", "", "Filter on number of forks")
+	cmd.Flags().StringVar(&opts.Query.Qualifiers.GoodFirstIssues, "good-first-issues", "", "Filter on number of issues with the 'good first issue' label")
+	cmd.Flags().StringVar(&opts.Query.Qualifiers.HelpWantedIssues, "help-wanted-issues", "", "Filter on number of issues with the 'help wanted' label")
+	cmdutil.StringSliceEnumFlag(cmd, &opts.Query.Qualifiers.In, "in", "", nil, []string{"name", "description", "readme"}, "Restrict search to specific part of repository")
+	cmd.Flags().StringSliceVar(&opts.Query.Qualifiers.Language, "language", nil, "Filter based on the coding language")
+	cmd.Flags().StringSliceVar(&opts.Query.Qualifiers.License, "license", nil, "Filter based on license type")
+	cmdutil.NilBoolFlag(cmd, &opts.Query.Qualifiers.Mirror, "mirror", "", "Filter based on mirror state")
+	cmd.Flags().StringVar(&opts.Query.Qualifiers.Org, "org", "", "Filter on organization")
+	cmd.Flags().StringVar(&opts.Query.Qualifiers.Pushed, "updated", "", "Filter on last updated at date")
+	cmd.Flags().StringVar(&opts.Query.Qualifiers.Repo, "repo", "", "Filter on repository name")
+	cmd.Flags().StringVar(&opts.Query.Qualifiers.Size, "size", "", "Filter on a size range, in kilobytes")
+	cmd.Flags().StringVar(&opts.Query.Qualifiers.Stars, "stars", "", "Filter on number of stars")
+	cmd.Flags().StringSliceVar(&opts.Query.Qualifiers.Topic, "topic", nil, "Filter on topic")
+	cmd.Flags().StringVar(&opts.Query.Qualifiers.Topics, "number-topics", "", "Filter on number of topics")
```

Comment:
Would we consider naming the `--forks` flag `--number-forks` for consistency with this one? On the other hand, there are many more filters that take in a number expression, so following the same logic the should all be prefixed with `--number`. This might be excessive (i.e. too verbose). What do you think?

Reply:
The main reason I made this flag `number-topics` is because the qualifiers of `topic` and `topics` was too confusing and I wanted to differentiate them. There are two fork qualifiers `fork` and `forks` and in that case I changed `fork` to `include-forks`. 

I think it is too verbose to add `--number` to beginning of all the number qualifiers and I think it is a bit of a misnomer since all the number qualifiers are actually allowed to be ranges in ES.

I am open to changing the `--topic` flag to something else and then changing `--number-topics` back to `--topics` if you think that is more consistent.

Reply:
I'm fine with the flags staying named the way they are right now. This was mostly food for thought.

If we wanted to simplify the surface area, we could consider consolidating the two flags into one:
```
gh search repos --fork=only
gh search repos --fork=false
gh search repos --fork=">5"

gh search repos --topic=gh-extension
gh search repos --topic="<2"
```
But overloaded functionality like this might be ultimately confusing.